File: summer2024/hw9/31609560_Problem1.py
import time
import requests
from pymongo import MongoClient
from bs4 import BeautifulSoup

url = "https://finance.yahoo.com/most-active"

response = requests.get(url)

if(response.status_code != 200):
	print("Request failed. Trying again")
	response = requests.get(url)
	if (response.status_code != 200):
		print("Request failed. Exiting")
		quit()

soup = BeautifulSoup(response.text, "html.parser")


num = 0

while True:
	data = soup.find_all("tr",{"class":"simpTblRow"})
	query = []
	
	for i in data:
		symbol = i.find("a", {"class":"Fw(600)"}).string
		name = i.find("td", {"aria-label":"Name"}).string
		price = float(i.find("td", {"aria-label":"Price (Intraday)"}).string)
		change = float(i.find("td", {"aria-label":"Price (Intraday)"}).string)
		volume = round(float((i.find("td", {"aria-label":"Volume"}).string).replace('M', '')), 2)
	
		query.append({"Symbol":symbol, "Name":name, "Price (Intraday)":price, "Change":change, "Volume":volume})	
		
	client = MongoClient()
	db = client['stocks']
	collection = db['info']
		
	collection.insert_many(query)
	print("Scraping data")
	time.sleep(180)

File: summer2024/hw9/31609560_Problem1.php
<?php
require 'vendor/autoload.php';
$client = new MongoDB\Client();
$db = $client->stocks;
$collection = $db->info;

$result = $collection->find([]);

?>
<!DOCTYPE html>
<html>

<body>
    <center>
    <table border="1">
        <tr>
            <th><a href="symbol.php">Symbol</a></th>
            <th><a href="name.php">Name</a></th>
            <th><a href="price.php">Price (Intraday)</a></th>
            <th><a href="change.php">Change</a></th>
            <th><a href="volume.php">Volume (Millions)</a></th>
        </tr>
        <?php
        foreach ($result as $doc) {
            echo "<tr>";
            foreach ($doc as $key => $value) {
                if ($key != "_id") {
                    echo "<td> {$value} </td>";
                }
            }
            echo "<tr>";
        }
        ?>
    </table>
    </center>
</html>
